yt_dlp/extractor/redgifs.py

```diff
@@ -0,0 +1,98 @@
+# coding: utf-8
+from .common import InfoExtractor
+from ..compat import compat_parse_qs
+from ..utils import ExtractorError, int_or_none
+
+
+def _parse_gif_entry(gif_data):
+    video_id = gif_data["id"]
+    title = " ".join(gif_data.get('tags') or [])
+    formats = [
+        {
+            "format_id": "gif",
+            "url": gif_data["urls"]["gif"]
+        },
+        {
+            "format_id": "sd",
+            "url": gif_data["urls"]["sd"]
+        },
+        {
+            "format_id": "hd",
+            "url": gif_data["urls"]["hd"]
+        }
+    ]
+    webpage_url = "https://redgifs.com/watch/{}".format(video_id)
+
+    return {
+        'id': video_id,
+        'title': title,
+        'age_limit': 18,
+        'formats': formats,
+        'timestamp': gif_data.get('createDate'),
+        'view_count': int_or_none(gif_data.get('views')),
+        'uploader': gif_data.get('userName'),
+        'webpage_url': webpage_url,
+        'tags': gif_data.get('tags')
+    }
+
+
+class RedGifsSearchIE(InfoExtractor):
+    IE_DESC = 'Redgifs search'
+    _VALID_URL = r'https?://(?:www\.)?redgifs\.com/browse\?(?P<query>.*)'
+    _TESTS = [
+        {
+            'url': 'https://www.redgifs.com/browse?tags=Lesbian',
+            'info_dict': {
+                'id': 'tags=Lesbian',
+                'title': 'Lesbian',
+                'description': 'Redgifs search for Lesbian, ordered by trending'
+            },
+            'playlist_mincount': 19,
+        },
+        {
+            'url': 'https://www.redgifs.com/browse?type=g&order=latest&tags=Lesbian',
+            'info_dict': {
+                'id': 'type=g&order=latest&tags=Lesbian',
+                'title': 'Lesbian',
+                'description': 'Redgifs search for Lesbian, ordered by latest'
+            },
+            'playlist_mincount': 19,
+        }
+    ]
+
+    def _real_extract(self, url):
+        query_str = self._match_valid_url(url).group("query")
+
+        query = compat_parse_qs(query_str)
+        if not query.get('tags'):
+            raise ExtractorError('Invalid query tags', expected=True)
+
+        tags = query.get("tags")[0]
+        order = query.get("order", ("trending",))[0]
+        page = query.get("page", (1,))[0]
```

Comment:
maybe try adding an example of this to the tests? it should also just paginate between all pages imo but not one to judge

Reply:
(was referring to page query)

Reply:
Yeah, I wasn't sure whether to paginate through all pages, or just do the specified page, to be honest. 
Is there a standard? 
I've added a test of page query in the meanwhile

Reply:
If there is no page number given in the URL, get all pages. Otherwise, (optionally) you can get just the page in the URL. Note that when paginating, it is best to return a `PagedList` instead of a generator/list (and I think this extractor is able to do that)

Reply:
Sounds like a great tactic. Thanks for the heads-up on `PagedList`, implemented that now